trufflesecurity/trufflehog
# Detector-Competition-Feat: Added Vagrant Cloud Access Token Detector

### Description:
This PR adds detector for VagrantCloud Personal Access Token.
Vagrant => Single workflow to build and manage virtual machine environments
Website: https://www.vagrantup.com/


### Category: Developer tool

### Docs
link: https://developer.hashicorp.com/vagrant/vagrant-cloud/api/v2

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [X] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### Running Tests
<img width="753" alt="Screenshot 2023-10-22 at 7 03 58 PM" src="https://github.com/trufflesecurity/trufflehog/assets/8335453/455bce73-841c-45e2-bdf8-533eec982d00">

### How to test locally
1. Register/Sign up for Vagrant Cloud account: https://app.vagrantup.com/
2. Goto User Settings > Security > Authentication Tokens and Add description and click Generate Token

Check following image for reference (*Note*: The token shown in image is invalid, as it is already been deleted. )



![Screenshot 2023-10-22 at 6 55 32 PM](https://github.com/trufflesecurity/trufflehog/assets/8335453/6c8d1d54-3246-45ad-81ef-a88beebd527f)
